test/fw/ptl/utils/pbs_testsuite.py

```diff
@@ -911,10 +912,13 @@ def _revert_pbsconf_comm(self, primary_server, vals_to_set):
                     if val is None:
                         self.logger.error("Couldn't set %s in pbs.conf" %
                                           (conf))
-                        del(new_pbsconf[conf])
+                        keys_to_delete.append(new_pbsconf[conf])
                     else:
                         new_pbsconf[conf] = val
 
+            for key in keys_to_delete:
+                del(new_pbsconf[key])
```

Comment:
I don't understand this.  How is this different than deleting the keys as you see them?

Reply:
Python gets mad at you if you delete elements from a dict while iterating on it:
RuntimeError: dictionary changed size during iteration

Reply:
Now that you say this, it makes perfect sense.  Changing the list/dict you are iterating sounds like a really bad idea.